### ISSUE TITLE
Server Internal Error on creating/saving as draft

### ISSUE THREAD

**Issue by DanielFreira** [wagtail/wagtail#12280]
### Issue Summary

At times, saving or creating a page as a draft throws a Server Internal Error and then it publishes said page, bypassing worklows.

### Steps to Reproduce

1. Create or save a page;
2. Save it as draft.

### Other info

I can't replicate it on my development environment, which I assume the problem lies with the apache server hosting the wagtail project.
The only confusing part is that on apache error logs, the error is about a duplicate slug. The were no pages on the parent page which make me believe the page is getting saved/created twice hence the error. However, at the end it publishes the page.
For instance, if a user which is locked behind a workflow saves a page as a draft and triggers said error, it will instantly publish the page.
It happens quite commonly but not always and doesn't look like it has a pattern.

### Technical details

-   Python version: 3.10.12.
-   Django version: 4.2.13.
-   Wagtail version: 5.2.5.
-   Browser version: Google Chrome.
-   Apache version: 2.4.52
-   SO: Ubuntu Server 22.04.4

### Apache Vhost config

```
<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerName ******

    Header always set Strict-Transport-Security "max-age=31536000;"

    WSGIDaemonProcess portal python-path=/var/www/portal/**** python-home=/var/www/portal/portal-env
    WSGIProcessGroup portal
    WSGIScriptAlias / /var/www/portal/***/portal/wsgi.py

    Alias /static/ /var/www/portal/***/static/
    Alias /media/ /var/www/portal/***/media/

    <Directory /var/www/portal/***/static/>
        Options -Indexes
    	AllowOverride None
	Require all granted
    </Directory>

    <Directory /var/www/portal/***/media/>
        Options -Indexes
    	AllowOverride None
	Require all granted
    </Directory>

    <Directory /var/www/portal/***/portal>
        <Files wsgi.py>
            Require all granted
        </Files>
    </Directory>

    <Location "/admin/">
	Require ip ***
	Require ip ***
	Require all denied
    </Location>

    SSLEngine on
    SSLCertificateFile /etc/ssl/certs/***
    SSLCertificateKeyFile /etc/ssl/***
    
</VirtualHost>
</IfModule>
```

### Logs

The only logs I get is from apache:

```
[Tue Sep 03 00:00:19.611391 2024] [mpm_event:notice] [pid 1081:tid 140375681218432] AH00489: Apache/2.4.52 (Ubuntu) OpenSSL/3.0.2 mod_wsgi/4.9.0 Python/3.10 configured -- resuming normal operations
[Tue Sep 03 00:00:19.611458 2024] [core:notice] [pid 1081:tid 140375681218432] AH00094: Command line: '/usr/sbin/apache2'
[Tue Sep 03 09:53:18.697815 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] --- Logging error ---
[Tue Sep 03 09:53:18.700800 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] Traceback (most recent call last):
[Tue Sep 03 09:53:18.700844 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.700847 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.700850 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
[Tue Sep 03 09:53:18.700852 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = wrapped_callback(request, *callback_args, **callback_kwargs)
[Tue Sep 03 09:53:18.700855 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/views/decorators/cache.py", line 62, in _wrapper_view_func
[Tue Sep 03 09:53:18.700857 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = view_func(request, *args, **kwargs)
[Tue Sep 03 09:53:18.700860 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/urls/__init__.py", line 173, in wrapper
[Tue Sep 03 09:53:18.700862 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return view_func(request, *args, **kwargs)
[Tue Sep 03 09:53:18.700865 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/auth.py", line 165, in decorated_view
[Tue Sep 03 09:53:18.700867 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = view_func(request, *args, **kwargs)
[Tue Sep 03 09:53:18.700869 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/views/generic/base.py", line 104, in view
[Tue Sep 03 09:53:18.700871 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return self.dispatch(request, *args, **kwargs)
[Tue Sep 03 09:53:18.700874 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 132, in dispatch
[Tue Sep 03 09:53:18.700876 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return super().dispatch(request)
[Tue Sep 03 09:53:18.700879 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/views/generic/base.py", line 143, in dispatch
[Tue Sep 03 09:53:18.700881 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return handler(request, *args, **kwargs)
[Tue Sep 03 09:53:18.700883 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 145, in post
[Tue Sep 03 09:53:18.700885 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return self.form_valid(self.form)
[Tue Sep 03 09:53:18.700888 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 161, in form_valid
[Tue Sep 03 09:53:18.700897 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return self.save_action()
[Tue Sep 03 09:53:18.700900 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 183, in save_action
[Tue Sep 03 09:53:18.700902 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     self.parent_page.add_child(instance=self.page)
[Tue Sep 03 09:53:18.700904 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/treebeard/mp_tree.py", line 1091, in add_child
[Tue Sep 03 09:53:18.700906 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return MP_AddChildHandler(self, **kwargs).process()
[Tue Sep 03 09:53:18.700909 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/treebeard/mp_tree.py", line 393, in process
[Tue Sep 03 09:53:18.700911 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     newobj.save()
[Tue Sep 03 09:53:18.700914 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/usr/lib/python3.10/contextlib.py", line 79, in inner
[Tue Sep 03 09:53:18.700916 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return func(*args, **kwds)
[Tue Sep 03 09:53:18.700918 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/models/__init__.py", line 1438, in save
[Tue Sep 03 09:53:18.700920 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     self.full_clean()
[Tue Sep 03 09:53:18.700923 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/models/__init__.py", line 1389, in full_clean
[Tue Sep 03 09:53:18.700925 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     super().full_clean(*args, **kwargs)
[Tue Sep 03 09:53:18.700928 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/db/models/base.py", line 1502, in full_clean
[Tue Sep 03 09:53:18.700929 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     raise ValidationError(errors)
[Tue Sep 03 09:53:18.700938 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] django.core.exceptions.ValidationError: {'slug': ["The slug 'simple-test' is already in use within the parent page at '*path*'"], 'path': ['P\xc3\xa1gina com este Path j\xc3\xa1 existe.']}
[Tue Sep 03 09:53:18.700942 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] 
[Tue Sep 03 09:53:18.700943 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] During handling of the above exception, another exception occurred:
[Tue Sep 03 09:53:18.700945 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] 
[Tue Sep 03 09:53:18.700948 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] Traceback (most recent call last):
[Tue Sep 03 09:53:18.700954 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/usr/lib/python3.10/logging/__init__.py", line 1103, in emit
[Tue Sep 03 09:53:18.700956 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     stream.write(msg + self.terminator)
[Tue Sep 03 09:53:18.700961 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] UnicodeEncodeError: 'ascii' codec can't encode character '\\xe1' in position 2994: ordinal not in range(128)
[Tue Sep 03 09:53:18.700969 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] Call stack:
[Tue Sep 03 09:53:18.703239 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
[Tue Sep 03 09:53:18.703246 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = self.get_response(request)
[Tue Sep 03 09:53:18.703250 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/base.py", line 140, in get_response
[Tue Sep 03 09:53:18.703252 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = self._middleware_chain(request)
[Tue Sep 03 09:53:18.703255 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703257 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703259 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703261 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703264 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703266 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703269 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703270 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703273 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703275 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703278 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703280 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703282 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703284 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703287 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703289 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703291 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703301 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703305 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703307 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703309 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703311 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703314 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703316 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703319 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703320 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703323 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703325 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703328 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703330 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703332 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703334 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703337 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703339 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703342 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/deprecation.py", line 134, in __call__
[Tue Sep 03 09:53:18.703343 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response or self.get_response(request)
[Tue Sep 03 09:53:18.703346 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703348 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703351 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/allauth/account/middleware.py", line 14, in __call__
[Tue Sep 03 09:53:18.703355 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = self.get_response(request)
[Tue Sep 03 09:53:18.703358 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703360 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703362 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/corsheaders/middleware.py", line 56, in __call__
[Tue Sep 03 09:53:18.703364 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     result = self.get_response(request)
[Tue Sep 03 09:53:18.703367 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 57, in inner
[Tue Sep 03 09:53:18.703369 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = response_for_exception(request, exc)
[Tue Sep 03 09:53:18.703372 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 143, in response_for_exception
[Tue Sep 03 09:53:18.703374 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     log_response(
[Tue Sep 03 09:53:18.703377 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/utils/log.py", line 241, in log_response
[Tue Sep 03 09:53:18.703378 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     getattr(logger, level)(
[Tue Sep 03 09:53:18.703386 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] Message: '%s: %s'
[Tue Sep 03 09:53:18.703388 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] Arguments: ('Internal Server Error', '/admin/pages/add/generic/genericpage/2785/')
[Tue Sep 03 09:53:18.703414 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] ERROR Internal Server Error: /admin/pages/add/generic/genericpage/2785/
[Tue Sep 03 09:53:18.703416 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] Traceback (most recent call last):
[Tue Sep 03 09:53:18.703418 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
[Tue Sep 03 09:53:18.703420 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = get_response(request)
[Tue Sep 03 09:53:18.703421 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
[Tue Sep 03 09:53:18.703423 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = wrapped_callback(request, *callback_args, **callback_kwargs)
[Tue Sep 03 09:53:18.703425 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/views/decorators/cache.py", line 62, in _wrapper_view_func
[Tue Sep 03 09:53:18.703427 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = view_func(request, *args, **kwargs)
[Tue Sep 03 09:53:18.703429 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/urls/__init__.py", line 173, in wrapper
[Tue Sep 03 09:53:18.703431 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return view_func(request, *args, **kwargs)
[Tue Sep 03 09:53:18.703434 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/auth.py", line 165, in decorated_view
[Tue Sep 03 09:53:18.703436 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     response = view_func(request, *args, **kwargs)
[Tue Sep 03 09:53:18.703438 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/views/generic/base.py", line 104, in view
[Tue Sep 03 09:53:18.703440 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return self.dispatch(request, *args, **kwargs)
[Tue Sep 03 09:53:18.703442 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 132, in dispatch
[Tue Sep 03 09:53:18.703443 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return super().dispatch(request)
[Tue Sep 03 09:53:18.703445 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/views/generic/base.py", line 143, in dispatch
[Tue Sep 03 09:53:18.703447 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return handler(request, *args, **kwargs)
[Tue Sep 03 09:53:18.703449 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 145, in post
[Tue Sep 03 09:53:18.703451 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return self.form_valid(self.form)
[Tue Sep 03 09:53:18.703452 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 161, in form_valid
[Tue Sep 03 09:53:18.703454 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return self.save_action()
[Tue Sep 03 09:53:18.703456 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/admin/views/pages/create.py", line 183, in save_action
[Tue Sep 03 09:53:18.703458 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     self.parent_page.add_child(instance=self.page)
[Tue Sep 03 09:53:18.703459 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/treebeard/mp_tree.py", line 1091, in add_child
[Tue Sep 03 09:53:18.703461 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return MP_AddChildHandler(self, **kwargs).process()
[Tue Sep 03 09:53:18.703463 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/treebeard/mp_tree.py", line 393, in process
[Tue Sep 03 09:53:18.703465 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     newobj.save()
[Tue Sep 03 09:53:18.703467 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/usr/lib/python3.10/contextlib.py", line 79, in inner
[Tue Sep 03 09:53:18.703468 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     return func(*args, **kwds)
[Tue Sep 03 09:53:18.703470 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/models/__init__.py", line 1438, in save
[Tue Sep 03 09:53:18.703472 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     self.full_clean()
[Tue Sep 03 09:53:18.703480 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/wagtail/models/__init__.py", line 1389, in full_clean
[Tue Sep 03 09:53:18.703482 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     super().full_clean(*args, **kwargs)
[Tue Sep 03 09:53:18.703484 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]   File "/var/www/portal/portal-env/lib/python3.10/site-packages/django/db/models/base.py", line 1502, in full_clean
[Tue Sep 03 09:53:18.703485 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146]     raise ValidationError(errors)
[Tue Sep 03 09:53:18.703487 2024] [wsgi:error] [pid 18321:tid 140375535728192] [remote ***.***.***.***:61146] django.core.exceptions.ValidationError: {'slug': ["The slug 'simple-test' is already in use within the parent page at '*path*'"], 'path': ['P\xc3\xa1gina com este Path j\xc3\xa1 existe.']}
```

**Comment by thibaudcolas** [comment]
Hi @DanielFreira, thank you for the report. I don’t think we see much issues specific to Apache, so this is surprising.

When you say this can’t be replicated in your dev environment – does that mean without Apache? Could you try running Apache on your dev environment, and come back with more information? Even better if you could set up our [bakerydemo](https://github.com/wagtail/bakerydemo) website with Apache and replicate it there.

> For instance, if a user which is locked behind a workflow saves a page as a draft and triggers said error, it will instantly publish the page.

Could you share more info about how the workflow is configured, and the permissions of the user in question?

---

Ultimately if it’s indeed Apache-specific I’d expect the fix will be a configuration change on your side, rather than a bug fix for Wagtail, but even then it’d still be interesting to know if there are any changes we should make to better protect against this kind of issue.

**Comment by DanielFreira** [comment]
Hi @thibaudcolas 

Apologies for the lack of information of the development side.
On the dev environment I simply run it through the inbuilt django development server, indeed without apache. That's why I'm thinking it being a server or service issue. However, I find it strange the error about duplicate slugs.

Any kind of user has triggered this anomaly. The workflows are quite simple, tasks being:
1- User creates/edits page
2- Manager approves
3- Publisher, approves and publishes

So what is happening is that users on step 1 that want to save a whole page or a change as a draft, will sometimes trigger this issue. If this happens to a user without publishing permissions it will result in a published page. 
To note that even If I as a superuser (administrator) that could bypass the workflow will trigger this if I save as a draft. (yet again, not always but quite common)

I will try to clone the server and run the demo (hopefully tomorrow) to see what happens.

**Comment by DanielFreira** [comment]
Hello there!

Sadly I couldn't allocate time to redo the server and test out the apache configuration.
However I did try the bakery demo with the django built-in development server on the production server and didn't see any issue.

While trying to solve another issue (sub-domains falling always on the same wagtail site), I swapped the runtime from apache to gunicorn and left apache as a reverse proxy as well as serving the static/media files. I still found this thread's issue with that configuration.
Since issue 2 was still unsolved I tried gunicorn and nginx. Issue 2 was still unsolved but eventually I solved it by deleting and creating a new site entry on wagtail admin. At this point, I tested out this thread's issue and could not replicate it.

As of now, my config is a wagtail multi-site running with gunicorn and using nginx to serve media/static files as well as reverse proxy.
Sadly I couldn't find the specific cause of this issue so it could be docummented.

Thx
